backend/packages/Upgrade/src/api/services/FeatureFlagService.ts

```diff
@@ -797,6 +797,7 @@ export class FeatureFlagService {
     logger.info({ message: 'Imported feature flags', details: createdFlags });
     return fileStatusArray;
   }
+
   public async validateImportFeatureFlags(
```

Comment:
Can you refactor this code? You are parsing the file twice once to get the `featureFlagsIds` and the second time to get the `validationErrors`. During the featureFlagIds parsing, you are not throwing any error; the second time, returning `INCOMPATIBLE`.
You should parse only once and use that result for further validation.

Reply:
code refractored